Zend/zend_compile.c

```diff
@@ -3778,6 +3778,8 @@ static uint32_t zend_compile_args(
 			arg_count++;
 		}
 
+		CG(arg_num) = arg_num;
```

Comment:
As mentioned off-list, this will break for `foo(bar(1, 2) | default)`. You'll need to back-up `CG(arg_num)` and restore it at the end of the loop.